main/main.pde

```diff
@@ -41,102 +40,92 @@ void setup()
-      for (int y = 0; y < kinect.height; y++) {
-        int offset = x + y * kinect.width;
-        int d = depth[offset];
-        if (d > minThresh && d < maxThresh) {
-          ellapsedTime = millis();
-          if(ready == true) {
-            ready = false;
-            for(int i = 0; i < 24; i++) {
-              system.addParticle(new PVector(random(x-50, x+50), random(y-50, y+50)));
-            }
-          }
-        } else {
-          if(millis() > ellapsedTime + 100) {
-            ready = true;
-          }
-        }
-      }
-    }
+    firstScreen(depth);
   } else {
+    secondScreen(depth);
+  } 
+}
+
+void firstScreen(int[] depth) {
+  system.update();
+  if(millis() > ellapsedTime + 2000) {
+    system.clearCount();
     background(0);
-    particleSystem.addParticle(new PVector(random(width), random(height)));
+    ready = true;
+  }
 
-    image(kinect.getDepthImage(), 640, 0);
+  // Update the particle system each frame
+  if(initialStart == true) {
+    initialStart = false;
+    delay(2000);
+  }
 
-    particleSystem.run();
-    float sumX = 0;
-    float sumY = 0;
-    float totalPixels = 0;
-    float avgX = 0;
-    float avgY = 0;
+  pixelParser(depth);
+}
 
-    int[] depth = kinect.getRawDepth();
+void secondScreen(int[] depth) {
+  background(0);
+  particleSystem.addParticle(new PVector(random(width), random(height)));
+  particleSystem.run();
+
+  float avgX = 0;
+  float avgY = 0;
+
+  sumX = 0;
+  sumY = 0;
+  totalPixels = 0;
+  
+  pixelParser(depth);
+
+  avgX = sumX / totalPixels;
+  avgY = sumY / totalPixels;
+  PVector avgPosition = new PVector(avgX, avgY);
+
+  if(totalPixels > 4500){
+    particleSystem.getAttracted(avgPosition);
+  } else if (totalPixels > 0 && totalPixels < 4000){
+    particleSystem.getRepulsed(avgPosition);
+  }
+}
 
-    for(int x = 0; x < kinect.width; x++){
-		  for(int y = 0; y < kinect.height; y++){
-			
-			  int offset = x + y * kinect.width;
-        int depthValue = depth[offset];
-        int minTrash = 500;
-        int maxTrash = 740;
+void pixelParser(int[] depth) {
+  for(int x = 0; x < kinect.width; x++){
+    for(int y = 0; y < kinect.height; y++){
+      int offset = x + y * kinect.width;
+      int d = depth[offset];
 
-        if(depthValue > minTrash && depthValue < maxTrash){
-          //img.pixels[offset] = color(255,0, 150);
+      if(d > minThresh && d < maxThresh) {
+        if (startScreenDone == false) {
+          addParticlesFirstScreen(d, x, y);
+        } else {
           sumX += x;
           sumY += y;
           totalPixels ++;
         }
-  		}
-	  }
-    avgX = sumX / totalPixels;
-    avgY = sumY / totalPixels;
-
-  	//img.updatePixels();
-	
-	  PVector avgPosition = new PVector(avgX, avgY);
-
-    if(totalPixels > 4500){
-      particleSystem.getAttracted(avgPosition);
-    } else if (totalPixels > 0 && totalPixels < 4000){
-      particleSystem.getRepulsed(avgPosition);
+      } else {
+        // print(millis(), ellapsedTime, "\n");
```

Comment:
do we still need this line?

Reply:
Removed here: 0f5152541bc7dae94776ca4dc74e2aaec2ed724a